experimental/list_formatter/src/list_formatter.rs

```diff
@@ -2,82 +2,47 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-use crate::provider::*;
+use crate::{provider::ListFormatterPatternsV1, ListStyle};
 use core::fmt::{self, Write};
 use icu_locid::Locale;
 use icu_provider::prelude::*;
 use writeable::*;
 
-/// Represents the type of a list. See the
-/// [CLDR spec](https://unicode.org/reports/tr35/tr35-general.html#ListPatterns)
-/// for an explanation of the different types.
-#[derive(Copy, Clone, PartialEq, Debug)]
-pub enum ListType {
-    /// A conjunction
-    And,
-    /// A disjunction
-    Or,
-    /// A list of units
-    Unit,
-}
-
-/// Represents the style of a list. See the
-/// [CLDR spec](https://unicode.org/reports/tr35/tr35-general.html#ListPatterns)
-/// for an explanation of the different styles.
-#[derive(Copy, Clone, PartialEq, Debug)]
-pub enum ListStyle {
-    /// A typical list
-    Wide,
-    /// A shorter list
-    Short,
-    /// The shortest type of list
-    Narrow,
-}
-
 /// A formatter that renders sequences of items in an i18n-friendly way. See the
 /// [crate-level documentation](crate) for more details.
-pub struct ListFormatter {
-    data: DataPayload<ListFormatterPatternsV1Marker>,
+pub struct ListFormatter<M: ResourceMarker<Yokeable = ListFormatterPatternsV1<'static>>> {
+    data: DataPayload<M>,
     style: ListStyle,
 }
 
-impl ListFormatter {
-    /// Creates a new [`ListFormatter`] from the given provider. The provider has to support the
-    /// correct key for the given type:
-    /// [`LIST_FORMAT_AND_V1`](crate::provider::key::LIST_FORMAT_AND_V1),
-    /// [`LIST_FORMAT_OR_V1`](crate::provider::key::LIST_FORMAT_OR_V1), or
-    /// [`LIST_FORMAT_UNIT_V1`](crate::provider::key::LIST_FORMAT_UNIT_V1).
-    /// An error is returned if the key or language are not available, or if there were any
-    /// deserialization errors.
-    pub fn try_new<T: Into<Locale>, D: DynProvider<ListFormatterPatternsV1Marker> + ?Sized>(
+impl<M: ResourceMarker<Yokeable = ListFormatterPatternsV1<'static>>> ListFormatter<M> {
+    /// Creates a new [`ListFormatter`] that produces a list for the given [`ResourceMarker`]. See
+    /// [`crate::markers`].
+    pub fn try_new<T: Into<Locale>, D: ResourceProvider<M> + ?Sized>(
```

Comment:
Thought: This is an interesting approach.

I'm not sure which I like better between this and the `try_new_[and/or/unit]` approach.

Pros of this approach:

- Reduced redundant information between the methods and the markers
- Looks very Rusty

Cons:

- Slightly less alignment between Rust and FFI; FFI still needs the separate functions
- Doesn't really save on any "typing" since `try_new::<And>` is longer than `try_new_and`
- Requires exporting (and importing) the extra `And` type

Neither pros nor cons:

- Could enable a client of ICU4X to make their own ResourceMarker satisfying the given trait bound